components/form/UploadImageInput.tsx

```diff
@@ -54,19 +56,44 @@ const UploadImageInput: React.FC<UploadImageProps> = ({
       const file = e.target.files[0];
 
       if (file) {
-        const imageName = uuidv4();
-        const imagePath = CDN + userId + '/' + imageName;
+        try {
+          const { data: listData, error: listError } = await supabase.storage
+            .from('images')
+            .list(userId, { recursive: true });
 
-        setImageSrc(imagePath);
-        setIsImageUploaded(true);
-        setError?.('');
+          if (listError) {
+            console.error('Error fetching old images:', listError);
+            return;
+          }
+
+          const oldImages = listData?.map((image) => image.name) || [];
```

Comment:
This all looks great but I'll share the same feedback I've given a few other contributors - let's lean on TypeScript and avoid using a fallback value (` || []`). 

If it TypeScript complains of the expected types it's preferable that we add an assertion wherever it makes sense to, but if the value coming out of any query is not what we expected we should let the code break to make it easier for us later if we need to debug.